From: csagan5 <32685696+csagan5@users.noreply.github.com>
Date: Sun, 17 Mar 2019 20:51:11 +0100
Subject: Always show home page settings

Remove the "Force Enable Home Page Button" flag, which was bugged, and always
display home page settings.
This allows removing the home page button and restoring it from settings.
---
 .../browser/init/ProcessInitializationHandler.java | 12 --------
 .../partnercustomizations/HomepageManager.java     | 17 ++---------
 .../preferences/ChromePreferenceManager.java       |  7 -----
 .../browser/preferences/MainPreferences.java       | 12 +++-----
 .../chrome/browser/util/FeatureUtilities.java      | 34 ----------------------
 chrome/browser/about_flags.cc                      |  3 --
 chrome/browser/android/chrome_feature_list.cc      |  4 ---
 chrome/browser/android/chrome_feature_list.h       |  1 -
 8 files changed, 7 insertions(+), 83 deletions(-)

diff --git a/chrome/android/java/src/org/chromium/chrome/browser/init/ProcessInitializationHandler.java b/chrome/android/java/src/org/chromium/chrome/browser/init/ProcessInitializationHandler.java
--- a/chrome/android/java/src/org/chromium/chrome/browser/init/ProcessInitializationHandler.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/init/ProcessInitializationHandler.java
@@ -356,18 +356,6 @@ public class ProcessInitializationHandler {
         deferredStartupHandler.addDeferredTask(new Runnable() {
             @Override
             public void run() {
-                if (HomepageManager.shouldShowHomepageSetting()) {
-                    RecordHistogram.recordBooleanHistogram("Settings.ShowHomeButtonPreferenceState",
-                            HomepageManager.isHomepageEnabled());
-                    RecordHistogram.recordBooleanHistogram("Settings.HomePageIsCustomized",
-                            !HomepageManager.getInstance().getPrefHomepageUseDefaultUri());
-                }
-            }
-        });
-
-        deferredStartupHandler.addDeferredTask(new Runnable() {
-            @Override
-            public void run() {
                 // Starts syncing with GSA.
                 AppHooks.get().createGsaHelper().startSync();
             }
diff --git a/chrome/android/java/src/org/chromium/chrome/browser/partnercustomizations/HomepageManager.java b/chrome/android/java/src/org/chromium/chrome/browser/partnercustomizations/HomepageManager.java
--- a/chrome/android/java/src/org/chromium/chrome/browser/partnercustomizations/HomepageManager.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/partnercustomizations/HomepageManager.java
@@ -83,10 +83,7 @@ public class HomepageManager {
      * @return Whether or not homepage is enabled.
      */
     public static boolean isHomepageEnabled() {
-        if (FeatureUtilities.isHomePageButtonForceEnabled()) {
-            return getInstance().getPrefHomepageEnabled();
-        }
-        return false;
+        return getInstance().getPrefHomepageEnabled();
     }
 
     /**
@@ -98,13 +95,6 @@ public class HomepageManager {
     }
 
     /**
-     * @return Whether or not homepage setting should be shown.
-     */
-    public static boolean shouldShowHomepageSetting() {
-        return FeatureUtilities.isHomePageButtonForceEnabled();
-    }
-
-    /**
      * @return Homepage URI string, if it's enabled. Null otherwise or uninitialized.
      */
     public static String getHomepageUri() {
@@ -118,8 +108,7 @@ public class HomepageManager {
     }
 
     /**
-     * @return The default homepage URI if the homepage is partner provided or the new tab page
-     *         if the homepage button is force enabled via flag.
+     * @return The default homepage URI.
      */
     public static String getDefaultHomepageUri() {
         return UrlConstants.NTP_NON_NATIVE_URL;
@@ -132,7 +121,7 @@ public class HomepageManager {
      * @see #isHomepageEnabled
      */
     public boolean getPrefHomepageEnabled() {
-        return mSharedPreferences.getBoolean(PREF_HOMEPAGE_ENABLED, true);
+        return mSharedPreferences.getBoolean(PREF_HOMEPAGE_ENABLED, false);
     }
 
     /**
diff --git a/chrome/android/java/src/org/chromium/chrome/browser/preferences/ChromePreferenceManager.java b/chrome/android/java/src/org/chromium/chrome/browser/preferences/ChromePreferenceManager.java
--- a/chrome/android/java/src/org/chromium/chrome/browser/preferences/ChromePreferenceManager.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/preferences/ChromePreferenceManager.java
@@ -154,13 +154,6 @@ public class ChromePreferenceManager {
     private static final String CHROME_MODERN_DESIGN_ENABLED_KEY = "chrome_modern_design_enabled";
 
     /**
-     * Whether or not the home page button is force enabled.
-     * Default value is false.
-     */
-    public static final String HOME_PAGE_BUTTON_FORCE_ENABLED_KEY =
-            "home_page_button_force_enabled";
-
-    /**
      * Whether or not the homepage tile will be shown.
      * Default value is false.
      */
diff --git a/chrome/android/java/src/org/chromium/chrome/browser/preferences/MainPreferences.java b/chrome/android/java/src/org/chromium/chrome/browser/preferences/MainPreferences.java
--- a/chrome/android/java/src/org/chromium/chrome/browser/preferences/MainPreferences.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/preferences/MainPreferences.java
@@ -206,15 +206,11 @@ public class MainPreferences extends PreferenceFragmentCompat
         updateSyncAndServicesPreference();
         updateSearchEnginePreference();
 
-        if (HomepageManager.shouldShowHomepageSetting()) {
-            Preference homepagePref = addPreferenceIfAbsent(PREF_HOMEPAGE);
-            if (FeatureUtilities.isNewTabPageButtonEnabled()) {
-                homepagePref.setTitle(R.string.options_startup_page_title);
-            }
-            setOnOffSummary(homepagePref, HomepageManager.getInstance().getPrefHomepageEnabled());
-        } else {
-            removePreferenceIfPresent(PREF_HOMEPAGE);
+        Preference homepagePref = addPreferenceIfAbsent(PREF_HOMEPAGE);
+        if (FeatureUtilities.isNewTabPageButtonEnabled()) {
+            homepagePref.setTitle(R.string.options_startup_page_title);
         }
+        setOnOffSummary(homepagePref, HomepageManager.getInstance().getPrefHomepageEnabled());
 
         if (NightModeUtils.isNightModeSupported() && FeatureUtilities.isNightModeAvailable()) {
             addPreferenceIfAbsent(PREF_UI_THEME);
diff --git a/chrome/android/java/src/org/chromium/chrome/browser/util/FeatureUtilities.java b/chrome/android/java/src/org/chromium/chrome/browser/util/FeatureUtilities.java
--- a/chrome/android/java/src/org/chromium/chrome/browser/util/FeatureUtilities.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/util/FeatureUtilities.java
@@ -73,7 +73,6 @@ public class FeatureUtilities {
     private static Boolean sHasGoogleAccountAuthenticator;
     private static Boolean sHasRecognitionIntentHandler;
 
-    private static Boolean sIsHomePageButtonForceEnabled;
     private static Boolean sIsHomepageTileEnabled;
     private static Boolean sIsNewTabPageButtonEnabled;
     private static Boolean sIsBottomToolbarEnabled;
@@ -178,7 +177,6 @@ public class FeatureUtilities {
     public static void cacheNativeFlags() {
         cacheCommandLineOnNonRootedEnabled();
         FirstRunUtils.cacheFirstRunPrefs();
-        cacheHomePageButtonForceEnabled();
         cacheHomepageTileEnabled();
         cacheNewTabPageButtonEnabled();
         cacheBottomToolbarEnabled();
@@ -228,38 +226,6 @@ public class FeatureUtilities {
         return Build.VERSION.SDK_INT > Build.VERSION_CODES.M;
     }
 
-    /**
-     * Cache whether or not the home page button is force enabled so on next startup, the value can
-     * be made available immediately.
-     */
-    public static void cacheHomePageButtonForceEnabled() {
-        if (PartnerBrowserCustomizations.isHomepageProviderAvailableAndEnabled()) return;
-        ChromePreferenceManager.getInstance().writeBoolean(
-                ChromePreferenceManager.HOME_PAGE_BUTTON_FORCE_ENABLED_KEY,
-                ChromeFeatureList.isEnabled(ChromeFeatureList.HOME_PAGE_BUTTON_FORCE_ENABLED));
-    }
-
-    /**
-     * @return Whether or not the home page button is force enabled.
-     */
-    public static boolean isHomePageButtonForceEnabled() {
-        if (sIsHomePageButtonForceEnabled == null) {
-            ChromePreferenceManager prefManager = ChromePreferenceManager.getInstance();
-
-            sIsHomePageButtonForceEnabled = prefManager.readBoolean(
-                    ChromePreferenceManager.HOME_PAGE_BUTTON_FORCE_ENABLED_KEY, false);
-        }
-        return sIsHomePageButtonForceEnabled;
-    }
-
-    /**
-     * Resets whether the home page button is enabled for tests. After this is called, the next
-     * call to #isHomePageButtonForceEnabled() will retrieve the value from shared preferences.
-     */
-    public static void resetHomePageButtonForceEnabledForTests() {
-        sIsHomePageButtonForceEnabled = null;
-    }
-
     private static void cacheServiceManagerForDownloadResumption() {
         boolean resumptionDownloadInReducedMode =
                 ChromeFeatureList.isEnabled(ChromeFeatureList.SERVICE_MANAGER_FOR_DOWNLOAD);
diff --git a/chrome/browser/about_flags.cc b/chrome/browser/about_flags.cc
--- a/chrome/browser/about_flags.cc
+++ b/chrome/browser/about_flags.cc
@@ -1932,9 +1932,6 @@ const FeatureEntry kFeatureEntries[] = {
     {"enable-chrome-duet-labels", flag_descriptions::kChromeDuetLabelsName,
      flag_descriptions::kChromeDuetLabelsDescription, kOsAndroid,
      FEATURE_VALUE_TYPE(chrome::android::kChromeDuetLabeled)},
-    {"force-enable-home-page-button", flag_descriptions::kHomePageButtonName,
-     flag_descriptions::kHomePageButtonDescription, kOsAndroid,
-     FEATURE_VALUE_TYPE(chrome::android::kHomePageButtonForceEnabled)},
     {"enable-ntp-button", flag_descriptions::kNtpButtonName,
      flag_descriptions::kNtpButtonDescription, kOsAndroid,
      FEATURE_VALUE_TYPE(chrome::android::kNTPButton)},
diff --git a/chrome/browser/android/chrome_feature_list.cc b/chrome/browser/android/chrome_feature_list.cc
--- a/chrome/browser/android/chrome_feature_list.cc
+++ b/chrome/browser/android/chrome_feature_list.cc
@@ -143,7 +143,6 @@ const base::Feature* kFeaturesExposedToJava[] = {
     &kEphemeralTab,
     &kExploreSites,
     &kHandleMediaIntents,
-    &kHomePageButtonForceEnabled,
     &kHomepageTile,
     &kHorizontalTabSwitcherAndroid,
     &kIdentityDisc,
@@ -439,9 +438,6 @@ const base::Feature kForegroundNotificationManager{
 const base::Feature kHandleMediaIntents{"HandleMediaIntents",
                                         base::FEATURE_ENABLED_BY_DEFAULT};
 
-const base::Feature kHomePageButtonForceEnabled{
-    "HomePageButtonForceEnabled", base::FEATURE_DISABLED_BY_DEFAULT};
-
 const base::Feature kHomepageTile{"HomepageTile",
                                   base::FEATURE_ENABLED_BY_DEFAULT};
 
diff --git a/chrome/browser/android/chrome_feature_list.h b/chrome/browser/android/chrome_feature_list.h
--- a/chrome/browser/android/chrome_feature_list.h
+++ b/chrome/browser/android/chrome_feature_list.h
@@ -77,7 +77,6 @@ extern const base::Feature kEphemeralTab;
 extern const base::Feature kExploreSites;
 extern const base::Feature kForegroundNotificationManager;
 extern const base::Feature kHandleMediaIntents;
-extern const base::Feature kHomePageButtonForceEnabled;
 extern const base::Feature kHomepageTile;
 extern const base::Feature kHorizontalTabSwitcherAndroid;
 extern const base::Feature kIdentityDisc;
-- 
2.11.0

